how-to-release.md

```diff
@@ -0,0 +1,274 @@
+Each release should be executed by a Release Manager, who is selected/proposed by the steering committee. This document describes the process that the Release Manager follows to perform a release. Any changes to this process should be discussed and adopted on the dev@ mailing list.
+
+This release process should be strictly followed by Pravega and also followed by all the ecosystem projects inside Pravega organization by default, but Pravega ecosystem projects can have its own release instruction page that points to this one as a reference and then explains what is different.
```

Comment:
I have addressed @andreipaduroiu comments(https://lists.cncf.io/g/cncf-pravega-dev/message/20) in this commit [d8e4bf0](https://github.com/pravega/.github/pull/19/commits/d8e4bf0689270b31334cbe9b2f45504678f4cf03). I took most of the advice here. here is still some thoughts.
> We should also add verbiage to say that any exception to this process will need to be vetted by the SC using a "Release process change proposal" (such as this one).

The definition of "exception" here is a bit ambiguous. Does that mean the difference of the general idea or the detailed steps.
For example, 
For projects do not have snapshot published, all the complication for the `-SNAPSHOT` removal is not there.
For go-based projects like operators most of the commands here does not apply.
For the rust client, they have some extra steps like building and publishing python wheels.
For connectors, we could skip the vote cycle and the docker image push step.

That is also why I chose the six projects which can apply this model. I can also support like giving some kind of freedom to the ecosystem projects that they have extend this document and do some adjustment accordingly. But given the diversity of the projects, I don't think it is easy to do that each exception requires a complete review through the SC.